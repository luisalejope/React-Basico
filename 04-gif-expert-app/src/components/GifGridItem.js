import React from 'react'
import PropTypes from 'prop-types'

const GifGridItem = ({id, title, url}) => {
    return (
        <>
        <img src={url} alt={title}></img>
        </>
    )
}

GifGridItem.propTypes = {

}

export default GifGridItem
