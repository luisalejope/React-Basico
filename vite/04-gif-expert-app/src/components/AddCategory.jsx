import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({ onAddCategory }) => {

    const [inputValue, setInputValue] = useState('')
    const onInputChange = (e) => setInputValue(e.target.value)
    

    const onSubmit = (e) => {
        e.preventDefault()

        if (inputValue.trim().length <= 1) {
            return;
        }

        onAddCategory(inputValue.trim())
        setInputValue('')
    }


  return (
    <form onSubmit={onSubmit} aria-label="form">  
        <input type="text" placeholder="Buscar Gifs" onChange={onInputChange} value={inputValue}/>
    </form>
  )
}


AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired
}

