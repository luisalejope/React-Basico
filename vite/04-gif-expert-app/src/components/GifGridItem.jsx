import PropTypes from 'prop-types'

export const GifGridItem = ({title, url, id}) => {
  return (
    <div className='card' id={id}>
        <h2>
            {title.trim() !== '' ? title : 'GIF SIN TITULO'}
        </h2>
        <img src={url} alt={title} />
    </div>
  )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    id: PropTypes.string
}
