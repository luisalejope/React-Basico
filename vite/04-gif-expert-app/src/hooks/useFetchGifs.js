import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";


export const useFetchGifs = (category) => {
    
    const [gifs, setGifs] = useState([])
    const [isLoading, setIsLoading] = useState(false)

    const requestGifs = async() => {
        try {
            setIsLoading(true)
            const res = await getGifs(category)
            setGifs(res)
            console.log(res)
        } catch (error) {
            console.error(error)
        } finally {
            setIsLoading(false)
        }
    }

    useEffect(() => {
        requestGifs()
    }, [category])
    

    return {
        gifs,
        isLoading
    }
}
