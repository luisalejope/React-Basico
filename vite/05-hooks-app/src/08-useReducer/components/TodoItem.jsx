
import PropTypes from 'prop-types'

export const TodoItem = ({todo, handleDeleteTodo, handleUpdateTodo}) => {

    const onUpdateStatus = () => {
        handleUpdateTodo({
            ...todo,
            done: !todo.done
        })
    }

    return (
        <li className="list-group-item d-flex justify-content-between align-items-center" onClick={onUpdateStatus}>
            <span className={todo.done ? 'text-decoration-line-through': ''}>{todo.desc}</span>
            <button className="btn btn-danger" onClick={()=>handleDeleteTodo(todo.id)} >Eliminar</button>
        </li>
    )
}

TodoItem.propTypes = {
    todo: PropTypes.object.isRequired,
    handleDeleteTodo: PropTypes.func.isRequired,
    handleUpdateTodo: PropTypes.func.isRequired,
}

