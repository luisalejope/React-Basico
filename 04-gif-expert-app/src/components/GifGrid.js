import React from "react";
import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
// import GifGridItem from "./GifGridItem";
// import { getGifs } from "../helpers/getGifs";

const GifGrid = ({ category }) => {
  // const [images, setImages] = useState([]);
  const {loading} = useFetchGifs();
  /* useEffect(() => {
    getGifs(category)
        .then(setImages);
  }, [category]); */

  return (
    <>
      <h3>{category}</h3>
        {loading ? 'Cargando...' : 'Data cargada'}
      {/* <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div> */}
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
