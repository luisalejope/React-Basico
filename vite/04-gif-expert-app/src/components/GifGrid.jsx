import PropTypes from 'prop-types'
import { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'


export const GifGrid = ({ category }) => {

    const {gifs, isLoading} = useFetchGifs(category)
    
    return (
        <>
            <h3>{category}</h3>
            <div className='card-grid'>
                { isLoading ? <p>Loading...</p> : gifs.map( gif => (
                    <GifGridItem key={gif.id} {...gif} />
                ))}
            </div>
        </>
    )
}
/* 4hJUgNoRRsiqFKOAr3qnnLRzwwkh0dZM */
GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
