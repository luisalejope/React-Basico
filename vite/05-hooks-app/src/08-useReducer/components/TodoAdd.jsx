import PropTypes from 'prop-types'
import { useForm } from '../../hooks/useForm'

export const TodoAdd = ({ handleSubmit }) => {
    const { todoInput, onInputChange, onResetForm } = useForm({
        todoInput: '',
    })

    const onAddTodo = (e) => {
        e.preventDefault()

        if (todoInput === '') return;

        handleSubmit(todoInput)
        onResetForm();
    }

    return (
        <form>
            <input type="text" className="form-control" name='todoInput' onChange={onInputChange} value={todoInput} />
            <button type="submit" className="btn btn-primary mt-2" onClick={onAddTodo}>Agregar</button>
        </form>
    )
}

TodoAdd.propTypes = {
    handleSubmit: PropTypes.func.isRequired
}