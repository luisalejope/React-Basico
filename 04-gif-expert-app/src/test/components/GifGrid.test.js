import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en <GifGrid />", () => {
  const category = "soy una categoria";
  
  test("debe de mostrarse correctamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar items cuando se cargan imagenes useFetchGifs", () => {
    const gifs =[{
        id: 'abc',
        title: 'titulo a',
        url: 'https://localhost/had/cosa.jpg'
    }]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot()
  });
});
