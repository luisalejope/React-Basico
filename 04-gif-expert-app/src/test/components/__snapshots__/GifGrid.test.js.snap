// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    soy una categoria
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="titulo a"
      url="https://localhost/had/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    soy una categoria
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
